File: gb.py
def filter_short_strings(array):
    result = []
    for string in array:
        if len(string) <= 3:
            result.append(string)
    return result

# Ввод массива строк с клавиатуры
n = int(input("Введите количество строк в массиве: "))
input_array = []
print("Введите строки:")
for _ in range(n):
    input_array.append(input())

# Формирование нового массива
result_array = filter_short_strings(input_array)

# Вывод результата
print("Новый массив строк, длина которых <= 3 символам:")
for string in result_array:
    print(string)
```

### Примеры использования

1. Для массива `["Hello", "2", "world", ":-)"]`:
    - Ввод: `["Hello", "2", "world", ":-)"]`
    - Вывод: `["2", ":-)"]`

2. Для массива `["1234", "1567", "-2", "computer science"]`:
    - Ввод: `["1234", "1567", "-2", "computer science"]`
    - Вывод: `["-2"]`

3. Для массива `["Russia", "Denmark", "Kazan"]`:
    - Ввод: `["Russia", "Denmark", "Kazan"]`
    - Вывод: `[]`

File: gb.java
java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Ввод массива строк с клавиатуры
        System.out.println("Введите количество строк в массиве:");
        int n = scanner.nextInt();
        scanner.nextLine(); // Потребление новой строки после nextInt()

        String[] inputArray = new String[n];
        System.out.println("Введите строки:");
        for (int i = 0; i < n; i++) {
            inputArray[i] = scanner.nextLine();
        }

        // Формирование нового массива
        String[] resultArray = filterShortStrings(inputArray);

        // Вывод результата
        System.out.println("Новый массив строк, длина которых <= 3 символам:");
        for (String str : resultArray) {
            System.out.println(str);
        }
    }

    public static String[] filterShortStrings(String[] array) {
        int count = 0;

        // Подсчет количества строк, длина которых <= 3 символам
        for (String str : array) {
            if (str.length() <= 3) {
                count++;
            }
        }

        // Создание нового массива нужного размера
        String[] result = new String[count];
        int index = 0;

        // Заполнение нового массива
        for (String str : array) {
            if (str.length() <= 3) {
                result[index++] = str;
            }
        }

        return result;
    }
}
